Chatkit/Redux/Reducers/Model/RoomListReducer.swift

```diff
@@ -0,0 +1,145 @@
+
+extension Reducer.Model {
+    
+    struct RoomList: Reducing {
+        
+        // MARK: - Types
+        
+        typealias ActionType = Action
+        typealias StateType = RoomListState
+        typealias DependenciesType = NoDependencies
+        
+        // MARK: - Reducer
+        
+        static func reduce(action: ActionType, state: StateType, dependencies: DependenciesType) -> StateType {
+            if let action = action as? InitialStateAction {
+                return self.reduce(action: action, state: state, dependencies: dependencies)
+            }
+            else if let action = action as? AddedToRoomAction {
+                return self.reduce(action: action, state: state, dependencies: dependencies)
+            }
+            else if let action = action as? RemovedFromRoomAction {
+                return self.reduce(action: action, state: state, dependencies: dependencies)
+            }
+            else if let action = action as? RoomUpdatedAction {
+                return self.reduce(action: action, state: state, dependencies: dependencies)
+            }
+            else if let action = action as? RoomDeletedAction {
+                return self.reduce(action: action, state: state, dependencies: dependencies)
+            }
+            else if let action = action as? ReadStateUpdatedAction {
+                return self.reduce(action: action, state: state, dependencies: dependencies)
+            }
+            
+            return state
+        }
+        
+        // MARK: - Private reducers
+        
+        private static func reduce(action: InitialStateAction, state: StateType, dependencies: DependenciesType) -> StateType {
+            let readSummaries = action.event.readStates.reduce(into: [String : ReadSummaryState]()) {
+                $0[$1.roomIdentifier] = ReadSummaryState(unreadCount: $1.unreadCount)
+            }
+            
+            let rooms = action.event.rooms.reduce(into: [String : RoomState]()) {
+                let readSummary = readSummaries[$1.identifier] ?? .empty
+                
+                $0[$1.identifier] = RoomState(identifier: $1.identifier,
+                                              name: $1.name,
+                                              isPrivate: $1.isPrivate,
+                                              pushNotificationTitle: $1.pushNotificationTitleOverride,
+                                              customData: $1.customData,
+                                              lastMessageAt: $1.lastMessageAt,
+                                              readSummary: readSummary,
+                                              createdAt: $1.createdAt,
+                                              updatedAt: $1.updatedAt)
+            }
```

Comment:
This would be a lot more readable with named args....
```
let readSummaries = action.event.readStates.reduce(into: [String : ReadSummaryState]()) { dict, readSummaryState in
    dict[readSummaryState.roomIdentifier] = ReadSummaryState(unreadCount: readSummaryState.unreadCount)
}

let rooms = action.event.rooms.reduce(into: [String : RoomState]()) { dict, roomState in
    let readSummary = readSummaries[roomState.identifier] ?? .empty
    
    dict[roomState.identifier] = RoomState(identifier: roomState.identifier,
                                           name: roomState.name,
                                           isPrivate: roomState.isPrivate,
                                           pushNotificationTitle: roomState.pushNotificationTitleOverride,
                                           customData: roomState.customData,
                                           lastMessageAt: roomState.lastMessageAt,
                                           readSummary: readSummary,
                                           createdAt: roomState.createdAt,
                                           updatedAt: roomState.updatedAt)
}
```

Reply:
This is a matter o personal preference.